src/Tree.cc

```diff
@@ -4,31 +4,36 @@
 
 #include <assert.h>
 #include <algorithm>
+#include <functional>
 #include <sstream>
-#include <boost/regex.hpp>
+#include <tuple>
 
 Tree::~Tree()
 {
-	this->nodecache.clear();
-	this->nodeidcache.clear();
+	this->nodecachemap.clear();
+	this->nodeidmap.clear();
 }
 
 /*!
 	Returns the cached string representation of the subtree rooted by \a node.
 	If node is not cached, the cache will be rebuilt.
 */
-const std::string &Tree::getString(const AbstractNode &node, const std::string &indent) const
+const std::string Tree::getString(const AbstractNode &node, const std::string &indent) const
 {
 	assert(this->root_node);
-	if (!this->nodecache.contains(node)) {
-		this->nodecache.clear();
-		this->nodeidcache.clear();
-		NodeDumper dumper(this->nodecache, indent, false);
+	bool idString = false;
+	bool idPrefix = false;
+
+	// Retrieve a nodecache given a tuple of NodeDumper constructor options
+	NodeCache &nodecache = this->nodecachemap[std::make_tuple(indent,idString,idPrefix)];
```

Comment:
I _think_ you should be able to just do `{indent,idString,idPrefix}`

Reply:
compiler doesn't like it
```C++
/home/hans/projects/openscad/src/Tree.cc: In member function ‘const string Tree::getString(const AbstractNode&, const string&) const’:
/home/hans/projects/openscad/src/Tree.cc:28:70: error: converting to ‘std::map<std::tuple<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, bool>, NodeCache>::key_type {aka std::tuple<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, bool>}’ from initializer list would use explicit constructor ‘constexpr std::tuple< <template-parameter-1-1> >::tuple(_UElements&& ...) [with _UElements = {const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, bool&, bool&}; <template-parameter-2-2> = void; _Elements = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool, bool}]’
  NodeCache &nodecache = this->nodecachemap[{indent,idString,idPrefix}];
```

Reply:
That's curious.. Don't think we need to dig deeper for now though :)